src/plots/plots.js

```diff
@@ -281,13 +281,19 @@ var extraFormatKeys = [
 //   is a list of all the transform modules invoked.
 //
 plots.supplyDefaults = function(gd) {
-    var oldFullLayout = gd._fullLayout || {},
-        newFullLayout = gd._fullLayout = {},
-        newLayout = gd.layout || {};
+    var oldFullLayout = gd._fullLayout || {};
 
-    var oldFullData = gd._fullData || [],
-        newFullData = gd._fullData = [],
-        newData = gd.data || [];
+    if(oldFullLayout._skipSD) {
+        delete oldFullLayout._skipSD;
+        return;
```

Comment:
my hack to prevent running `supplyDefaults` twice in full-replot update pathways.

Reply:
good call -> `_skipDefaults` in https://github.com/plotly/plotly.js/pull/2341/commits/1074804c79f23779bb94571283e2954a23d27af6